doc/porting/arch.rst

```diff
@@ -458,3 +458,104 @@ Each architecture also needs its own linker script, even if most sections can
+
+* On MMU systems, this guard area is an entire page whose permissions in the
+  page table will generate a fault on writes. This page needs to be
+  configured in the arch's _new_thread() function.
+
+* On MPU systems, one of the MPU regions needs to be reserved for the thread
+  stack guard area, whose size should be minimized. The region in the MPU
+  should be reconfigured on context switch such that the guard region
+  for the incoming thread is not writable.
+
+User Mode Threads
+=================
+
+To support user mode threads, hardware stack protection must be enabled and
+several kernel-to-arch APIs need to be implemented. Please see the
+documentation for each of these functions for more details:
+
+* :cpp:func:`_arch_buffer_validate()` to test whether the current thread has
+  access permissions to a particular memory region
+
+* :cpp:func:`_arch_user_mode_enter()` which will irreversably drop a supervisor
+  thread to user mode privileges. The stack must be wiped.
+
+* :cpp:func:`_arch_syscall_oops()` which generates a kernel oops when system
+  call parameters can't be validated, in such a way that the oops appears to be
+  generated from where the system call was invoked in the user thread
+
+* :cpp:func:`_arch_syscall_invoke0()` through
+  :cpp:func:`_arch_syscall_invoke6()` invoke a system call with the
+  appropriate number of arguments which must all be passed in during the
+  privilege elevation via registers.
+
+* :cpp:func:`_arch_is_user_context()` return nonzero if the CPU is currently
+  running in user mode
+
+* :cpp:func:`_arch_mem_domain_max_partitions_get()` which indicates the max
+  number of regions for a memory domain. MMU systems have an unlimited amount,
+  MPU systems have constraints on this.
+
+* :cpp:func:`_arch_mem_domain_partition_remove()` Remove a partition from
+  a memory domain if the currently executing thread was part of that domain.
+
+* :cpp:func:`_arch_mem_domain_destroy()` Reset the thread's memory domain
+  configuration
+
+In addition to implementing these APIs, there are some other tasks as well:
+
+* :cpp:func:`_new_thread()` needs to spawn threads with :c:macro:`K_USER` in
+  user mode
+
+* On context switch, the outgoing thread's stack memory should be marked
+  inaccessible to user mode. The incoming thread's stack memory should be
+  marked as accessible. This ensures that threads can't mess with other thread
+  stacks.
+
+* On context switch, the system needs to switch between memory domains for
+  the incoming and outgoing threads.
+
+* Thread stack areas must include a kernel stack region. This should be
+  inaccessible to user threads at all times. This stack will be used when
+  system calls are made. This should be fixed size for all threads, and must
+  be large enough to handle any system call.
+
+* A software interrupt or some kind of privilege elevation mechanism needs to
+  be established. This is closely tied to how the _arch_syscall_invoke macros
```

Comment:
can you add the reference tags to the docs for these macros (as you've done for others)

Reply:
unfortunately this is a set of 10 macros, not just an individual one. how can I refer to a family of macros?

Reply:
nevermind :)

Reply:
you might be able to drop an anchor on the first one and treat it like a group, with a link to said anchor